web/client/utils/CatalogUtils.js

```diff
@@ -64,6 +64,63 @@ const getThumb = (dc) => {
     }));
 };
 
+// Extract the relevant information from the wms URL for (RNDT / INSPIRE)
+const extractWMSParamsFromURL = wms => {
+    const params = new URLSearchParams(wms.value);
+    const lowerCaseParams = new URLSearchParams();
+    for (const [name, value] of params) {
+        lowerCaseParams.append(name.toLocaleLowerCase(), value);
+    }
+    const layerName = lowerCaseParams.get('layers');
+    const wmsVersion = lowerCaseParams.get('version');
+    if (layerName) {
+        return {
+            ...wms,
+            protocol: 'OGC:WMS',
+            name: layerName,
+            value: `${wms.value.match( /[^\?]+[\?]+/g)}SERIVCE=WMS${wmsVersion && `&VERSION=${wmsVersion}`}`
+        };
+    }
+    return false;
+};
+
+const getMetaDataDownloadFormat = (protocol = '', value = '') => {
+    let format = '';
+    const formatsMap = [
+        {
+            protocol: /www-download/g,
+            value: /shp\/.*\.zip/g,
```

Comment:
I like the final effect but unfortunately parsing the URL to guess the format is not a good practice, in general. If we have a protocol like WMS, we can expect the parameters "layers" etc..., but with this, we can not. Also having WMS in the URL is not synonymous of a WMS service. 
So this map should have, when `name` attribute is not present, only to check the protocol: 

- protocol = "http://www.opengis.net/def/serviceType/ogc/wms" --> displayValue: WMS
- protocol = "http://www.opengis.net/def/serviceType/ogc/wfs"  --> displayName: WFS
- protocol = "https://registry.geodati.gov.it/metadata-codelist/ProtocolValue/www-download" --> displayName: "Download"
- protocol = "http://www.w3.org/TR/xlink/" --> displayName "Link"
- default "Link"

I liked to have "shapeFile", kmz too, but we cannot assume anything from the URL.

Reply:
@offtherailz , could `protocol = "http://www.w3.org/TR/xlink/" --> displayName "Link"` be omitted as the display value will be `Link`, the default one?

```
    const formatsMap = [
        {
            protocol: 'https://registry.geodati.gov.it/metadata-codelist/ProtocolValue/www-download',
            displayValue: 'Download'
        },
        {
            protocol: 'http://www.opengis.net/def/serviceType/ogc/wms',
            displayValue: 'WMS'
        },
        {
            protocol: 'http://www.opengis.net/def/serviceType/ogc/wfs',
            displayValue: 'WFS'
        },
    ];
```